aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,118 @@
+By [Luke Latham](https://github.com/GuardRex) and [John Luo](https://github.com/JunTaoLuo)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/caching/middleware/sample)
+
+This document provides details on how to configure the Response Caching Middleware in ASP.NET Core applications. The middleware determines when responses are cacheable using headers, options, and features configured by the application, stores responses, and serves responses from cache. For an introduction to HTTP caching and the `ResponseCache` attribute, see [Response Caching](response.md).
+
+## Package
+To include the middleware in your project, add a reference to the  [`Microsoft.AspNetCore.ResponseCaching`](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package. The middleware is available for projects that target `.NETFramework 4.5.1` or `.NETStandard 1.3` or higher.
+
+## Extensions
+In `ConfgureServices`, add the middleware to your service collection.
+
+[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet1)]
+
+Configure the application to use the middleware when processing requests. The sample application adds a `Cache-Control` header to the response that will cache cacheable responses for up to 10 seconds. The sample also sends a `Vary` header to configure the cache to serve the response only if the `Accept-Encoding` header of subsequent requests matches that from the original request.
+
+[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet2)]
+
+The position of this middleware relative to other middleware in the pipeline is important. Any terminal middleware placed before this middleware will prevent the Response Caching Middleware from caching or serving the response. For example, if you place [Static File Middleware](xref:fundamentals/static-files) before this middleware, your static files will not be cached by the middleware. If you place Static File Middleware after this middleware, your static files will be cached.
+
+>[!NOTE]
+> If you place Static File Middleware before Response Caching Middleware, you can provide a prefix to the middleware that will force it to ignore requests with a path that doesn't match your prefix. For example, you can prefix your application's static file links with **assets** (`http://www.myapp.com/assets/staticfile.txt`) and configure the middleware to ignore any requests that do not include the **assets** prefix, `app.UseStaticFiles("/assets")`. Note that "**assets**" is only part of the path in your links to the static files. Your application will not have an **assets** folder, and your static files should remain in the **wwwroot** folder of your deployed application (**wwwroot\staticfile.txt**).
+
+> The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
```

Comment:
Fixed by dropping the first paragraph. This is a good, normal piece of juicy information on the behavior of the middleware. Let's try placing it within the body text there and see how that reads.